lib/message.rb

```diff
@@ -0,0 +1,15 @@
+require './lib/morse_translator.rb'
+
+class Message
+  attr_reader :text, :translator
+
+  def initialize(text, translator)
+    @text  = text
+    @translator   = translator
+  end
+
+  def translation
+    morse_translator.translate(@text)
```

Comment:
Close here. `translate` is a method on `MorseTranslator`, but you're holding your instance of `MorseTranslator` in the instance variable `@translator` here. In order to make this work you need to reference that instance variable.

Reply:
I've been hesitant to overuse instance variables in an effort to encapsulate my programs. I'm close, though, to understanding these scope issues.